### PR TITLE
handling for multistage dockerfiles

### PR DESCRIPTION
**Pull request by sosiouxme**

### CODE REVIEW

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 1]
```diff
@@ -54,7 +60,32 @@ def __init__(self, tasker, workflow, parent_registry=None, parent_registry_insec
...
+        base_image_str = str(self.workflow.builder.base_image)
+        nonce = 0
+        for parent in sorted(self.workflow.builder.parent_images.keys()):
+            image = ImageName.parse(parent)
+            if parent == base_image_str:
+                base_image = self._resolve_base_image(build_json)
+                image = self._ensure_image_registry(base_image)
+
+            if self.check_platforms:
+                self._validate_platforms_in_image(image)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_pull_base_image.py]: In theory, we only need to do this for the last stage in build.

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 2]
```diff
@@ -54,7 +60,32 @@ def __init__(self, tasker, workflow, parent_registry=None, parent_registry_insec
...
             self.parent_registry_insecure = False
 
-    def resolve_base_image(self, build_json):
+    def run(self):
+        """
+        Pull parent images and retag them uniquely for this build.
+        """
+        build_json = get_build_json()
+        base_image_str = str(self.workflow.builder.base_image)
+        nonce = 0
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_pull_base_image.py]: Consider using `enumerate` instead.

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 1 above)
Comment (lcarva) [reply atomic_reactor/plugins/pre_pull_base_image.py]: That's not true at all. Deleting this comment.

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 2 above)
Comment (sosiouxme) [reply atomic_reactor/plugins/pre_pull_base_image.py]: Thanks, makes sense.

    for nonce, parent in enumerate(sorted(self.workflow.builder.parent_images.keys())):

**Pull request closed without merging**